include/dosbox.h

```diff
@@ -73,9 +64,23 @@ enum SVGACards {
 }; 
 
 extern SVGACards svgaCard;
-extern MachineType machine;
 extern bool mono_cga;
 
+enum MachineType {
+	// In age-order: Hercules is the oldest and VGA is the newest
+	MCH_HERC  = 1 << 0,
+	MCH_CGA   = 1 << 1,
+	MCH_TANDY = 1 << 2,
+	MCH_PCJR  = 1 << 3,
+	MCH_EGA   = 1 << 4,
+	MCH_VGA   = 1 << 5,
+};
+
+extern MachineType machine;
+
+inline bool is_machine(const int type) {
+	return machine & type;
+}
```

Comment:
Same, I Don't think this change belongs in this PR?

Reply:
It make theachine enum 'bit-capable'; and like the bitops, the scope of the machine type covers the entire codebase. 

I figured it was sufficiently related that the two could go together.

Reply:
One option is to reword the PR title to suggest that the PR tries to put some foundation elements in place to better supybit operations (both in a general sense, but also DOSBox-specific sense; ie: the machine enum).

Reply:
My concern is there may be some hidden usage of these enums that rely on the implicit order/int values.

I mean, probably not, but having seen some of the codebase, I also wouldn't be entirely surprised...

Reply:
(yup.. I grepped every use-case our of the code base and made sure there weren't any value-assumptions or directional comparisons).  
But if there's an issue regardless (and I fully agree with your concern), the bisection will land on this commit whether it's in this PR or we put it in another PR.

Reply:
What's the rationale for making these enums bit-capable in the first place? It's not like you can set multiple machine types at once can you?

Reply:
It's visually easier to see:

```
machine_is(MCH_CGA | MCH_HERC l MCH_TANDY)
```

Very easy to read versus
```
If (machine == MCH_CGA || machine == MCH_HERC || machine == MCH_TANDY)
```

(Yeah, it's very minor)

Reply:
I've been thinking about using C++17's fold expressions to create a function to do this kind of comparison.

An example would look like `if ( any(machine,  MCH_CGA, MCH_HERC, MCH_TANDY) ) {}`, and because of templates, would work with any comparible type.

Reply:
That would be even more readable :+1: 

Ideally we could make it `constexpr` too so the looping over N args to construct the final bit value would be performed compile-time and be equivalent at runtime to `(MCH_CGA | MCH_HERC | MCH_TANDY)`. 

(although yeah.. I realize all of this is micro-optimization land, but it would still be nice knowing we could use this stuff in tight loops, like where some of the machine checks are currently done in the video code).